lib/service/connect.go

```diff
@@ -76,6 +85,22 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 	}
 }
 
+func (process *TeleportProcess) compareVersions(resp *proto.PingResponse) error {
+	serverVersion, err := semver.NewVersion(resp.ServerVersion)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	teleportVersion, err := semver.NewVersion(teleport.Version)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if math.Abs(float64(serverVersion.Major-teleportVersion.Major)) > 1 {
```

Comment:
I think the supported version differences is 1 right? So 6.2.20 is okay with 7.3.4 but not 8.0.5.

Reply:
https://goteleport.com/docs/setup/operations/upgrading/#component-compatibility

Reply:
I think this is right, but im not quite sure -- 7 would be compatible with 6 and 8 but 8 would only be compatible with 7 and 9 right?